### ISSUE TITLE
Separate live environments

### ISSUE THREAD

**Issue by rhlsthrm** [connext/monorepo#3834]
## Background

This is the current setup.
- `staging` which is kicked off from `staging` (merged from main).
- `testnet-prod` which is kicked off from `prod` branch (merged from staging).
- `mainnet` which is kicked off from `prod` branch (merged from staging).

This causes problems when we need to "break" staging to test new protocol level upgrades or other big, breaking changes. In these cases, the whole pipeline is frozen until we are ready to promote staging to prod.

This is a proposal for a better design:
- `staging`: Kicked off from `staging` branch. Can be merged into from any feature branch to test changes and is breakable at any time.
- `testnet`: Kicked off from `testnet` branch. Merged from `main` branch directly when we are ready to test prod ready changes from main.
- `mainnet`: Kicked off from `prod` branch. Merged from `testnet` branch when we are ready to promote testnet changes into mainnet.

### Linked Issues & Documentation

- List of linked issues, external links, etc

## Definition Of Done

Environments match above spec.

## Tasks

- [ ] Create branches matching spec.
- [ ] Change CI job code to kick off infra updates based on spec.

**Comment by LayneHaber** [comment]
shouldn't testnet come from staging

**Comment by rhlsthrm** [comment]
> shouldn't testnet come from staging

No, because think about the case where staging is used to test new features that should not be in prod yet (like the current optimistic roots setup).

**Comment by LayneHaber** [comment]
named the branch `testnet-prod` to be pedantic
